### PR TITLE
Code Review #1

### PR DESCRIPTION
**Pull request by peetklecha**
### Assignee Tasks

* [ ] added unit tests (or none needed)
* [ ] written relevant docs (or none needed)
* [ ] referenced any relevant issues (or none exist)

### Guidelines

Please add a description of this Pull Request's motivation, scope, outstanding issues or potential alternatives, reasoning behind the current solution, and any other relevant information for posterity.

---

_Your PR Notes Here_

### CODE REVIEW

File: server/db/models/creditCards.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+const db = require('../db')
+const {ENUM, STRING} = require('sequelize')
+
+const CreditCard = db.define('creditCard', {
```
Comment (HypeDis) [inline server/db/models/creditCards.js:4]: this better not be plain text

File: server/db/models/index.js [hunk 2]
```diff
@@ -1,18 +1,31 @@
+const Brand = require('./brand')
+const {CreditCard, Merchant} = require('./creditCards')
+const Department = require('./department')
+const Product = require('./product')
+const Review = require('./review')
 const User = require('./user')
+const Image = require('./product-images')
```
Comment (HypeDis) [inline server/db/models/index.js:7]: dont store raw images on this server

File: server/db/models/index.js [hunk 3]
```diff
@@ -1,18 +1,31 @@
...
 
-/**
- * If we had any associations to make, this would be a great place to put them!
- * ex. if we had another model called BlogPost, we might say:
- *
- *    BlogPost.belongsTo(User)
- */
+CreditCard.belongsTo(User)
```
Comment (HypeDis) [inline server/db/models/index.js:9]: probably want to use a 3rd party integration

File: server/db/models/index.js [hunk 4]
```diff
@@ -1,18 +1,31 @@
...
+Product.belongsTo(Department)
+Department.hasMany(Product)
+Product.belongsTo(Brand)
+Brand.hasMany(Product)
+Review.belongsTo(User)
+User.hasMany(Review)
+Review.belongsTo(Product)
+Product.hasMany(Review)
+Product.hasMany(Image)
+Image.belongsTo(Product)
```
Comment (HypeDis) [inline server/db/models/index.js:21]: make pushing cart to master your priority

File: server/api/brandRoutes.js [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+  try {
+    const brand = await Brand.findByPk(id)
+    brand.update(req.body)
+    res.status(200).send(brand)
+  } catch (err) {
+    next(err)
+  }
+})
+
+router.delete('/:id', async (req, res, next) => {
```
Comment (HypeDis) [inline server/api/brandRoutes.js:77]: make sure your admin routes are protected

File: client/components/productDetail.js [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+    const id = this.props.match.params.id
+    this.props.readProduct(id)
+  }
+
+  averageReview(reviews) {
+    if (reviews.length === 1) {
+      return reviews[0].rating
+    } else {
+      return (
+        reviews.reduce((accum, cur) => accum + cur.rating, 0) / reviews.length
```
Comment (HypeDis) [inline client/components/productDetail.js:30]: look into sequelize virtual columns

File: server/api/productsRoute.js [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+  }
+})
+
+router.get('/:id', async (req, res, next) => {
+  const id = req.params.id
+  try {
+    const product = await Product.findByPk(id, {
+      include: [
+        {
+          model: Review
```
Comment (HypeDis) [inline server/api/productsRoute.js:40]: look into how to do an aggregate on a join

File: client/components/productDetail.js [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+    this.averageReview = this.averageReview.bind()
+  }
+
+  componentDidMount() {
+    const id = this.props.match.params.id
+    this.props.readProduct(id)
+  }
+
+  averageReview(reviews) {
+    if (reviews.length === 1) {
```
Comment (HypeDis) [inline client/components/productDetail.js:26]: check that reviews is initially an iterable

File: client/components/productDetail.js [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+      let product = this.props.product
+      const reviewAvg = averageReview(product.reviews)
+        ? averageReview(product.reviews).toFixed(1)
+        : 'First to Review'
+
+      return (
+        <div className="pageContainer">
+          <div className="productDetailContainer">
+            <div className="leftSection">
+              <div className="imgListContainer">
```
Comment (HypeDis) [inline client/components/productDetail.js:49]: create sub components

File: public/style.css [hunk 10]
```diff
@@ -1,21 +1,390 @@
...
+  align-items: center;
+}
+
+.imgListContainer img {
+  width: 150px;
+  cursor: pointer;
+}
+
+.prodDetailBigImg {
+  max-width: 100%;
```
Comment (HypeDis) [inline public/style.css:213]: remove fixed width

File: client/components/departments_slide.js [hunk 11]
```diff
@@ -0,0 +1,46 @@
+import React, {Component} from 'react'
+import {Link} from 'react-router-dom'
+import {Carousel} from 'react-bootstrap'
+
+const Departments_slide = () => {
```
Comment (HypeDis) [inline client/components/departments_slide.js:5]: wtf

Comment (robertpengcode) [reply client/components/departments_slide.js]: the name of the component and the file is changed.

**Pull request merged**